### ISSUE TITLE
Debug hang when using this extension

### ISSUE THREAD

**Issue by lcharette** [breeze2/vscode-phpstan#16]
VSCode version : 1.46.1
PHP version:  7.3.13
XDebug version:  v2.7.0
OS: MacOS 10.15.5

When I run debug at a breakpoint (with xdebug), VSCode debug panel hang. It might show the variables, but expanding a variable doesn't work. The spinner will stay there indefinitely. See screenshot below : 
<img width="334" alt="85951122-1da25200-b92f-11ea-8c79-50b4e623fa20" src="https://user-images.githubusercontent.com/2566513/86307435-c62d0c00-bbe4-11ea-9bbf-03b0c12c8287.png">

When this happens, I can't "stop" execution of the "run", and get this timeout notification. Whole php might hang after (can't run script at all) : 
<img width="452" alt="85951125-2004ac00-b92f-11ea-8251-2ec3e481267e" src="https://user-images.githubusercontent.com/2566513/86307439-c9c09300-bbe4-11ea-95e9-7b9cf20ba0d5.png">

I tested with different projects, with local xdebug / php installed on the host machine and in a Vagrant box, plus . It used to work a couple of weeks ago. 

**So far, I've isolated the issue with this extension. When I disable it, everything is fine.** 

(ref.: https://github.com/felixfbecker/vscode-php-debug/issues/412)

**Comment by lcharette** [comment]
I did a fresh install of VSCode using AppCleaner, works fine now ¯\\\_(ツ)\_/¯

**Comment by lcharette** [comment]
Spoke too soon, issue is back

**Comment by breeze2** [comment]
I found that `phpstan analyse` will be blocked when `vscode-php-debug` is running, and I may not be able to handle it.
<img width="452" alt="44C97574-2A36-4A5C-86B1-81652FF60F3C" src="https://user-images.githubusercontent.com/5266711/86559703-39d96b00-bf8f-11ea-9352-829248ec0d1f.png">

By the way, you can try to use [psalm](https://psalm.dev/) for php linting.

**Comment by Wosi** [comment]
I ran into the same issue and reported it here: https://github.com/xdebug/vscode-php-debug/issues/497

**Comment by lcharette** [comment]
> I ran into the same issue and reported it here: [xdebug/vscode-php-debug#497](https://github.com/xdebug/vscode-php-debug/issues/497)

FYI, I already had a ticket open on that repo : https://github.com/xdebug/vscode-php-debug/issues/412

**Comment by zobo** [comment]
I sadly don't have the time to do a full analysis, but I'm guessing you have a `xdebug.remote_autostart = 1` configured, instead of using a trigger to start debugging. Since phpstan is also written in php it's most probably using the same runtime, config, and Xdebug Zend extension.
I would suggest you explicitly disable remote debugging when executing phpstan from this VSCode extension. One way could be `php -dxdebug.remote_enable=0 phpstan.php...` . With Xdebu3 it would be `php -dxdebug.mode=off phpstan...`. Another option is setting an environment `XDEBUG_CONFIG=remote_enable=0` or `XDEBUG_CONFIG=mode=off`

**Comment by lcharette** [comment]
I played a bit with @zobo idea in an attempt to get this fixed. Obvisouly, having `xdebug.remote_autostart` disabled in the  global php config doesn't produced the error, but at the same time debug wasn't working, so that's that. 

> I would suggest you explicitly disable remote debugging when executing phpstan from this VSCode extension

That's easier said than done. Nevertheless, I hacked a bit around the extension source code. Even when adding the `XDEBUG_CONFIG=mode=off` env [here](https://github.com/breeze2/vscode-phpstan/blob/3fe2fe7cbb3abe8f926040f2acbf40451315cfd1/src/controller.ts#L255) resulted in the same issue.
